server/server.js

```diff
@@ -115,6 +116,33 @@ app.get(`${apiBaseUrl}/milestone-report`,
     return res.json({ data: result });
   }));
 
+// Get hired report
+app.get(`${apiBaseUrl}/milestone-report/csv/hired`,
+  keycloak.allowRolesMiddleware('ministry_of_health'),
+  asyncMiddleware(async (req, res) => {
+    res.attachment('report.csv');
+    const csvStream = csv.format({ headers: true });
+    csvStream.pipe(res);
```

Comment:
How does this work? Piping into the response, and when the csvStream is closed then it sends the response as a blob?

Reply:
It actually sends as it goes, not when it finishes, AFAI.

Reply:
Oh, interesting. I'm trying to wrap my head around how it would know when to actually return the response blob to the client, but if it works then it isn't a big deal.